apps/search/js/providers/grid_provider.js

```diff
@@ -21,8 +23,15 @@ DataGridProvider.prototype = {
 
   render: function(results) {
     results.forEach(function(config, index) {
+      if (this.grid.getLength() >= this.MAX_RESULTS) {
```

Comment:
Yeah I tried that and for some reason it was really unreliable. I can try again if you feel strongly about it?